internal/api/handler/update_promotion_policy_v1alpha1.go

```diff
@@ -0,0 +1,69 @@
+package handler
+
+import (
+	"context"
+
+	"github.com/bufbuild/connect-go"
+	"github.com/pkg/errors"
+	kubeerr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kubev1alpha1 "github.com/akuity/kargo/api/v1alpha1"
+	typesv1alpha1 "github.com/akuity/kargo/internal/api/types/v1alpha1"
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+type UpdatePromotionPolicyV1Alpha1Func func(
+	context.Context,
+	*connect.Request[svcv1alpha1.UpdatePromotionPolicyRequest],
+) (*connect.Response[svcv1alpha1.UpdatePromotionPolicyResponse], error)
+
+func UpdatePromotionPolicyV1Alpha1(
+	kc client.Client,
+) UpdatePromotionPolicyV1Alpha1Func {
+	return func(
+		ctx context.Context,
+		req *connect.Request[svcv1alpha1.UpdatePromotionPolicyRequest],
+	) (*connect.Response[svcv1alpha1.UpdatePromotionPolicyResponse], error) {
+		var policy kubev1alpha1.PromotionPolicy
+		switch {
+		case req.Msg.GetYaml() != "":
+			if err := yaml.Unmarshal([]byte(req.Msg.GetYaml()), &policy); err != nil {
+				return nil, connect.NewError(connect.CodeInvalidArgument, errors.Wrap(err, "invalid yaml"))
```

Comment:
need similar validation as we did in create to be safe

Reply:
I feel like I wouldn't go out of our way to validate that the namespace is a "project."

Reply:
Adding more context... the controller doesn't care. It watches all Stages and Promotions regardless of what their namespace looks like. I think the label is a minor convenience that makes the get projects api call return less noise.

Reply:
It doesn't matter for controller but it matters for UI as the UI only watches for specific namespaces as projects so it will miss out stages deployed to some non-project ns as far as I understand.

Reply:
For delete/update I doubt the UI will show those stages that aren't even in a valid project unless the user makes a custom api call themselves.